streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -481,10 +479,24 @@ private static int getCrossRackTrafficCost(final Set<TaskTopicPartition> topicPa
      */
     private static boolean canPerformRackAwareOptimization(final ApplicationState applicationState,
                                                            final AssignedTask.Type taskType) {
-        final String rackAwareAssignmentStrategy = applicationState.assignmentConfigs().rackAwareAssignmentStrategy();
+        final AssignmentConfigs assignmentConfigs = applicationState.assignmentConfigs();
+        final String rackAwareAssignmentStrategy = assignmentConfigs.rackAwareAssignmentStrategy();
         if (StreamsConfig.RACK_AWARE_ASSIGNMENT_STRATEGY_NONE.equals(rackAwareAssignmentStrategy)) {
+            LOG.warn("Rack aware task assignment optimization disabled: rack aware strategy was set to {}",
+                rackAwareAssignmentStrategy);
+            return false;
+        }
+
+        if (!assignmentConfigs.rackAwareTrafficCost().isPresent()) {
```

Comment:
nit (here and for the check below):

```suggestion
        if (assignmentConfigs.rackAwareTrafficCost().isEmpty()) {
```

Reply:
isEmpty is JDK 11 unfortunately

Reply:
what??????? ridiculous lol